inc/saiswitch.h

```diff
@@ -976,8 +990,117 @@ typedef enum _sai_switch_attr_t
      * @default false
      */
     SAI_SWITCH_ATTR_SWITCH_SHELL_ENABLE,
+    
+    /**
+     * @brief Handle for switch profile id.
+     *
+     * Use this to retrive the Key-Vlaue pairs as part of switch 
+     * initialization. 
+     * @type sai_uint32_t
+     * @flags CREATE_ONLY
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_SWITCH_PROFILE_ID,
+
+    /**
+     * @brief Device Information for switch initialization.
+     * 
+     * Hardware information format is based on SAI implementations by vendors.
+     * String is NULL terminated. Format is vendor specific. 
+     *   Example: Like PCI location, I2C adddress etc.
+     * In case of NULL, First NPU attached to CPU will be initialized.
+     * Single NPU case this attribute is optional.
+     *
+     * @type sai_s8_list_t
+     * @flags CREATE_ONLY
+     * @default empty
+     */
+    SAI_SWITCH_ATTR_SWITCH_HARDWARE_INFO,
+
+    /**
+     * @brief Vendor specific path name of the firmware to load.
+     * 
+     * @type sai_s8_list_t
+     * @flags CREATE_ONLY
+     * @default empty
+     */
+    SAI_SWITCH_ATTR_FIRMWARE_PATH_NAME,
+
+    /**
+     * @brief Set to switch initialization or connect to NPU/SDK.
+     *
+     * TRUE - Initialize switch/SDK. 
+     * FALSE - Connect to SDK. This will connects library to the initialized SDK.
+     * After this call the capability attributes should be ready for retrieval
+     * via sai_get_switch_attribute()
+     *
+     * @type bool
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_SWITCH_ATTR_INIT_SWITCH,
```

Comment:
@lguohan , While looking at previous connect API, API will allows notification as argument, 
is that means multiple applications can register of same notifications example FDB/port notification need by app1 and app2 . 
In this scenario we need to generate  2 call back notification to 2 applications for same FDB event. 
correct me if i am wrong? 

in this case, each notification event will go with switch object created with init=true or should generate events with switch object id create for connect app1 and connect app2.

if this is real scenario, then it is better to go with connect and disconnect API's and use switch id as input. Then SAI can replicate events and send to multiple applications.

@itaibaz - Consider above scenario, app1 and app2 are used connect and registered for FDB notification. Now app2 alone is disconnected how SAI will aware it should trigger notifications to app1 alone, because present disconnect does not have any args.    

I am not aware of background of connect/disconnect API's. Please let me know expected behavior based on that we can converge on above solution.